etc/kernel-launchers/python/scripts/launch_ipykernel.py

```diff
@@ -178,11 +180,15 @@ def _validate_port_range(port_range):
                 )
     except ValueError as ve:
         raise RuntimeError(
-            f"Port range validation failed for range: '{port_range}'.  Error was: {ve}"
+            "Port range validation failed for range: '{port_range}'.  Error was: {ve}".format(
```

Comment:
Why are we moving back to format? instead of f-string?

Reply:
Oh yea, I forgot to mention this too, any reason we're not using fstrings? They've been introduced in py 3.6 and previous versions of python are no longer supported. If you're doing this for backwards compatibility, you might be unintentionally creating maintenence hell for yourself!

Reply:
From the description:
> We also should try to support older python versions in the launcher, so I reverted the use of f-strings for now (as my YARN systems are using older versions of Python).

I felt that enterprises tend to (significantly) lag behind on support versions and not using `f-strings` in the launcher enables the use of Python < 3.5 (including 2.x, btw).  Since f-strings are purely for code readability, I didn't think it was that big a deal to not use them (once I figured out how to get the `pyupgrade` to ignore the launcher).

If others strongly believe we need `f-strings` over (significant) backward compatibility, I'm okay moving to `f-strings` in the python launcher.  I just felt this particular tradeoff was better served learning toward B/C.